binary_search/Java/BinarySearch.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * @author: shakeelsamsu
+ * https://github.com/shakeelsamsu
+ * Implementation of Binary Search in Java
+ * Returns the location of element x in a given array (if present), or otherwise returns -1
+ */
+class BinarySearch {
+    int binarysearch(int[] nums, int x) {
+        int i = 0;
+        int f = nums.length - 1;
+        while(i <= f) {
+            
+            int mid = i + (f - 1) / 2; //find the middle
+
+            //check the middle
+            if(nums[mid] == x) return mid;
+
+            //if x is greater than the mid, ignore the left
+            if(nums[mid] < x) i = mid + 1;
+            
```

Comment:
Line contains following spacing inconsistencies:
- Trailing whitespaces.

*Origin: SpaceConsistencyBear, Section: `all.pyjava`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpveusbzv7/binary_search/Java/BinarySearch.java
+++ b/tmp/tmpveusbzv7/binary_search/Java/BinarySearch.java
@@ -17,7 +17,7 @@
 
             //if x is greater than the mid, ignore the left
             if(nums[mid] < x) i = mid + 1;
-            
+
             //otherwise ignore the right
             else f = mid - 1;
         }
```